client/src/components/userProfile.jsx

```diff
@@ -1,72 +1,72 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
+import ReportPage from './reportPage.jsx'
 
-const UserProfile = ({ userData }) => {
-  const data = userData[0] || {};
-  let followStatus = '' || 'Loading';
-  let bolt = '';
-  let username = data.userName || '';
-  if (data.isFollowing === 'true') {
-    followStatus = 'Following';
-  } else {
-    followStatus = 'Follow';
+class UserProfile extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      show: false,
+    };
+    this.showModal = this.showModal.bind(this);
+    this.hideModal = this.hideModal.bind(this);
   }
 
-  if (data.pro === 'true') {
-    bolt = <i className="fas fa-bolt" />;
+  showModal() {
+    this.setState({ show: true });
   }
 
-  if (data.userName && data.userName.length > 13) {
-    username = data.userName.slice(0, 10).concat('...');
+  hideModal() {
+    this.setState({ show: false });
   }
 
-  if (data.isFollowing === 'false') {
-    followStatus = (
-      <button type="button" className="up-button-unfollow" tabIndex="0" title="Unfollow">
-    Following
-      </button>
-    );
-  } else {
-    followStatus = (
-      <button type="button" className="up-button-follow" tabIndex="0" title="Follow">
-    Follow
-      </button>
-    );
-  }
-
-  return (
-    <div className="up">
-      <img className="up-photo" src={data.profilePhoto} alt="Loading" />
-      <br />
-      <span className="up-username">
-        {username}
-      </span>
-      <span className="up-bolt">
-        {bolt}
-      </span>
-      <br />
-      <div className="up-count">
-        <span className="up-followcount">
-          <i className="fas fa-users" />
-          {data.followers}
+  render() {
+    const { userData, following, handleFollow } = this.props;
+    const data = userData || {};
+    const followStatus = following ? 'following' : 'follow';
+    const proBolt = data.pro === 'true' ? <i className="fas fa-bolt" /> : '';
+    const username = data.userName && data.userName.length > 13 ? data.userName.slice(0, 10).concat('...') : data.userName;
+    return (
+      <div className="up">
+        <img className="up-photo" src={data.profilePhoto} alt="Loading" />
+        <br />
+        <span className="up-username">
```

Comment:
Not sure if span is the best HTML tag for this, maybe try to choose something more semantically precise, like h[1-5]?